
syntax = "proto3";

package mathematics;

message LengthComputationResult {
  double length = 1;
}

message ScheduleLengthComputationRequest {
  string id = 1;
  repeated int32 coordinates = 2;
}

message ScheduleLengthComputationResponse {}

message LookupLengthRequest {
  string id = 1;
}

message LookupLengthResponse {
  double length = 1;
}

service Geometry {
  rpc ScheduleLengthComputation(ScheduleLengthComputationRequest)
      returns (ScheduleLengthComputationResponse) {}

  rpc LookupLength(LookupLengthRequest) returns (LookupLengthResponse) {}
}
